File: uniapp_flutter/lib/utils/media_type.dart

enum MediaType {
  IMAGE,
  VIDEO,
  AUDIO,
  TEXT,
  PDF,
  BLOB,
  OTHERS
}

class MediaTypeHelper {
  static int getValue(MediaType status){
    switch(status){
      case MediaType.IMAGE:
        return 1;
      case MediaType.VIDEO:
        return 2;
      case MediaType.AUDIO:
        return 3;
      case MediaType.TEXT:
        return 4;
      case MediaType.PDF:
        return 5;
      case MediaType.BLOB:
        return 6;
      case MediaType.OTHERS:
        return 7;
      default:
        return 0;
    }
  }
  static final Map<int, MediaType> valueToStateNameMap = new Map();
  final int value = 0;
  static createMap() {

    for (MediaType myEnum in MediaType.values) {
      valueToStateNameMap[MediaTypeHelper.getValue(myEnum)]= myEnum;
    }
  }

  static MediaType getStateByValue(int value) {
    if(valueToStateNameMap.isEmpty) {
      createMap();
    }
    return valueToStateNameMap[value];
  }
}

File: app/app/src/main/java/com/vassar/unifiedapp/utils/MediaType.java
package com.vassar.unifiedapp.utils;
import java.util.HashMap;
import java.util.Map;

public enum MediaType {

    IMAGE(0),
    VIDEO(1),
    AUDIO(2),
    TEXT(3),
    PDF(4),
    BLOB(5),
    OTHERS(6);


    private static final Map<Integer, MediaType> valueToTypeMap = new HashMap<>();
    private final Integer value;

    static {
        for (MediaType myEnum : values()) {
            valueToTypeMap.put(myEnum.getValue(), myEnum);
        }
    }

    private MediaType(final Integer newValue) {
        value = newValue;
    }

    public Integer getValue() {
        return value;
    }

    public static MediaType getTypeByValue(Integer value) {
        return valueToTypeMap.get(value);
    }
}
